pkg/controller/servicebindingrequest/binder.go

```diff
@@ -237,26 +239,39 @@ func (b *binder) updateSpecContainers(
 	if containers, err = b.updateContainers(containers); err != nil {
 		return nil, err
 	}
-	if err = unstructured.SetNestedSlice(obj.Object, containers, containersPath...); err != nil {
+	if err = unstructured.SetNestedSlice(obj.Object, containers, b.getContainersPath()...); err != nil {
 		return nil, err
 	}
 	return obj, nil
 }
 
+func (b *binder) getContainersPath() []string {
+	return strings.Split(b.sbr.Spec.ApplicationSelector.BindingPath.PodSpecPath.Containers, ".")
+}
+
+func (b *binder) getVolumesPath() []string {
+	return strings.Split(b.sbr.Spec.ApplicationSelector.BindingPath.PodSpecPath.Volumes, ".")
+}
+
+func (b *binder) getSecretFieldPath() []string {
+	return strings.Split(*b.sbr.Spec.ApplicationSelector.BindingPath.CustomSecretPath, ".")
+}
+
 // removeSpecContainers find and edit containers resource subset, removing bind related entries
 // from the object. It can return error on extracting data, editing steps and final editing of to be
 // returned object.
 func (b *binder) removeSpecContainers(
```

Comment:
I have created a separate task for this: https://github.com/redhat-developer/service-binding-operator/issues/601

Reply:
Before this PR, I admit this method's signature isn't the most straight-forward but it didn't have any side effects related to the given `obj`, but a side effect has been introduced by this PR since a mutation has been introduced here, so I find it appropriate to solve it in this PR.

Reply:
Let me know if this fix is fine. 669c7cc5dad963f6cae0dc91382ba74768e392c2

Reply:
See https://github.com/redhat-developer/service-binding-operator/pull/513/files#r469825678

Reply:
see the above link for the reply.